### PR TITLE
SAK-48700 Portal site nav favorites migration to pinned sites

### PR DESCRIPTION
**Pull request by ern**
https://sakaiproject.atlassian.net/browse/SAK-48700

### CODE REVIEW

**Review by ottenhoff — APPROVED** [review]
LGTM

File: portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java [hunk 1]
```diff
@@ -124,58 +129,121 @@ public int doPost(String[] parts, HttpServletRequest req,
...
+		List<String> pinnedSites = portalService.getPinnedSites(userId);
+		List<String> unPinnedSites =  portalService.getUnpinnedSites(userId);
+		List<String> combinedSiteIds = Stream.concat(pinnedSites.stream(), unPinnedSites.stream()).collect(Collectors.toList());
+
+		// when the user has no sites it is most likely their first login since pinning was introduced
+		if (combinedSiteIds.isEmpty()) {
+			log.debug("User has no pinned site data performing favorites migration for user [{}]", userId);
+			// look to the users favorites stored in preferences
+			List<String> favoriteSiteIds = props.getPropertyList(FAVORITES_PROPERTY);
+			if (favoriteSiteIds != null && !favoriteSiteIds.isEmpty()) {
```
Comment (jesusmmp) [inline portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java:154]: Use StringUtils.isBlank or StringUtils.isBlank

File: portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java [hunk 2]
```diff
@@ -124,58 +129,121 @@ public int doPost(String[] parts, HttpServletRequest req,
...
+							log.debug("Adding site [{}] from favorites to pinned sites for user [{}]", id, userId);
+							portalService.addPinnedSite(userId, id, true);
+							combinedSiteIds.add(id);
+						});
+			}
 
-		List<String> existingSiteIds = portalService.getPinnedSites();
-		existingSiteIds.addAll(portalService.getUserUnpinnedSites());
+			List<String> seenSiteIds = props.getPropertyList(SEEN_SITES_PROPERTY);
+			if (seenSiteIds != null && !seenSiteIds.isEmpty()) {
```
Comment (jesusmmp) [inline portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java:174]: Use StringUtils.isBlank or StringUtils.isBlank

File: portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java [hunk 3]
```diff
@@ -840,71 +786,47 @@ else if (iconType.equalsIgnoreCase("cl")){
...
 		PinnedSite pin = pinnedSiteRepository.findByUserIdAndSiteId(userId, siteId)
 			.orElseGet(() -> new PinnedSite(userId, siteId));
 
-		List<PinnedSite> pinned = pinnedSiteRepository.findByUserIdOrderByPosition(userId);
-		int position = pinned.size() > 0 ? pinned.get(pinned.size() - 1).getPosition() + 1 : 1;
+
+		int position = PinnedSite.UNPINNED_POSITION;
+		if (isPinned) {
+			List<PinnedSite> pinned = pinnedSiteRepository.findByUserIdOrderByPosition(userId);
+            position = !pinned.isEmpty() ? pinned.get(pinned.size() - 1).getPosition() + 1 : 1;
```
Comment (jesusmmp) [inline portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java:816]: Spacing

File: portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java (hunk 1 above)
Comment (ottenhoff) [reply portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java:154]: I think StringUtils is only for Strings, no?

Comment (jesusmmp) [reply portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java:154]: Oops... Yesterday I happened to see that it was String.

Comment (bgarciaentornos) [reply portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java:154]: Maybe CollectionUtils.isNotEmpty?

**Pull request merged**